apt/minimization/minimizer.py

```diff
@@ -316,74 +342,143 @@ def fit(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, y: Optional[DATA_PANDA
+                    if generalize_using_transform:
+                        generalized = self._generalize_from_tree(x_test, x_prepared_test, nodes, self.cells,
+                                                                 self._cells_by_id)
+                    else:
+                        generalized = self._generalize_from_generalizations(x_test, self.generalizations)
                     accuracy = self.estimator.score(ArrayDataset(self.encoder.transform(generalized), y_test))
                     print('Removed feature: %s, new relative accuracy: %f' % (removed_feature, accuracy))
 
             # self._cells currently holds the chosen generalization based on target accuracy
 
             # calculate iLoss
-            self._ncp = self._calculate_ncp(X_test, self._generalizations, feature_data)
+            X_test_dataset = ArrayDataset(x_test, features_names=self._features)
+            self._ncp_scores.fit_score = self.calculate_ncp(X_test_dataset, generalize_using_transform)
+            self._ncp_scores.generalizations_score = self.calculate_ncp(X_test_dataset, False)
 
         # Return the transformer
         return self
 
     def transform(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, features_names: Optional[list] = None,
                   dataset: Optional[ArrayDataset] = None):
-        """ Transforms data records to representative points.
+        """ Transforms data records to representative points. Also sets the transform_score in self.ncp.
 
         :param X: The training input samples.
         :type X: {array-like, sparse matrix}, shape (n_samples, n_features), optional
-        :param features_names: The feature names, in the order that they appear in the data. Can be provided when
-                               passing the data as ``X`` and ``y``
+        :param features_names: The feature names, in the order that they appear in the data. Should be provided when
+                               passing the data as ``X`` as a numpy array
         :type features_names: list of strings, optional
         :param dataset: Data wrapper containing the training input samples and the predictions of the original model
                         on the training data. Either ``X`` OR ``dataset`` need to be provided, not both.
         :type dataset: `ArrayDataset`, optional
         :return: Array containing the representative values to which each record in ``X`` is mapped, as numpy array or
                  pandas DataFrame (depending on the type of ``X``), shape (n_samples, n_features)
         """
+        transformed = self._inner_transform(X, features_names, dataset)
+        transformed_dataset = ArrayDataset(transformed, features_names=self._features)
+        self._ncp_scores.transform_score = self.calculate_ncp(transformed_dataset, True)
+        return transformed
 
+    def calculate_ncp(self, samples: ArrayDataset, generalize_using_transform: bool = True):
```

Comment:
Does the user still need to fill this `generalize_using_transform` ? 
I thought having NCPScores would enable us here to get the correct NCP score, since we could track the process.

Reply:
Yes, because some of the stages in the process (like fitting the generalizer) are still affected by this param.